framework/src/util/SharedLibraryException.cpp

```diff
@@ -0,0 +1,42 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "cppmicroservices/SharedLibraryException.h"
+
+namespace cppmicroservices {
+
+SharedLibraryException::~SharedLibraryException() = default;
+
+SharedLibraryException::SharedLibraryException(const std::error_code ec, const std::string& msg, const Bundle& origin)
+	: std::system_error(ec, msg), origin(origin)
+{
+}
+
+Bundle SharedLibraryException::GetBundle() const {
+  return origin;
+}
+
+void SharedLibraryException::SetBundle(const Bundle &bundle) {
+  origin = bundle;
+}
```

Comment:
not ideal. There shouldn't be a need to set the bundle after construction. It only complicates the design.

Reply:
agreed; removed.